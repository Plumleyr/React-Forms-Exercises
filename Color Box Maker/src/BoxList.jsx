import Box from "./Box";
import { useState } from "react";
import NewBoxFrom from "./NewBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBox = (box) => {
    let newBox = { ...box, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };

  const removeBox = (id) => {
    setBoxes(boxes.filter((box) => box.id !== id));
  };

  return (
    <div>
      <NewBoxFrom addBox={addBox} />
      <div>
        {boxes.map((box, idx) => {
          return (
            <Box
              removeBox={removeBox}
              id={box.id}
              width={box.width}
              height={box.height}
              backgroundColor={box.backgroundColor}
              data-testid={`box-${idx}`}
            />
          );
        })}
      </div>
    </div>
  );
};

export default BoxList;
