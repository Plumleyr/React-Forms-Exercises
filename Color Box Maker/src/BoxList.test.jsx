import { render, fireEvent, screen } from "@testing-library/react";
import BoxList from "./BoxList";

it("doesn't break on load", () => {
  render(<BoxList />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a box to the list", () => {
  render(<BoxList />);

  const widthInput = screen.getByPlaceholderText("width");
  const heightInput = screen.getByPlaceholderText("height");
  const backgroundColorInput = screen.getByPlaceholderText("backgroundColor");
  const createButton = screen.getByText("Create Box");

  fireEvent.change(widthInput, { target: { value: "100px" } });
  fireEvent.change(heightInput, { target: { value: "200px" } });
  fireEvent.change(backgroundColorInput, { target: { value: "blue" } });
  fireEvent.click(createButton);

  const box = screen.getByTestId("box-0");
  expect(box).toHaveStyle("width: 100px;");
  expect(box).toHaveStyle("height: 200px;");
  expect(box).toHaveStyle("background-color: rgb(0, 0, 255);");
});
