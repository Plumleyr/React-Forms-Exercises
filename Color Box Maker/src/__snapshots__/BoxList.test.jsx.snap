// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        backgroundColor:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Create Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
